codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HuffmanEncoder.java

```diff
@@ -115,14 +117,79 @@ public void encode(OutputStream out, byte[] data, int off, int len) throws IOExc
      * @param data the string literal to be Huffman encoded
      * @return the number of bytes required to Huffman encode <code>data</code>
      */
-    public int getEncodedLength(byte[] data) {
-        if (data == null) {
-            throw new NullPointerException("data");
+    public int getEncodedLength(CharSequence data) {
+        if (data instanceof AsciiString) {
+            AsciiString string = (AsciiString) data;
+            try {
+                string.forEachByte(encodedLengthProcessor);
+                return encodedLengthProcessor.length();
+            } catch (Exception e) {
+                PlatformDependent.throwException(e);
+                return -1;
+            } finally {
+                encodedLengthProcessor.len = 0;
+            }
+        } else {
+            return getEncodedLengthSlowPath(data);
         }
+    }
+
+    private int getEncodedLengthSlowPath(CharSequence data) {
         long len = 0;
-        for (byte b : data) {
-            len += lengths[b & 0xFF];
+        for (int i = 0; i < data.length(); i++) {
+            len += lengths[data.charAt(i) & 0xFF];
         }
         return (int) ((len + 7) >> 3);
     }
+
+    private final class EncodeProcessor implements ByteProcessor {
+        ByteBuf out;
+        private long current;
+        private int n;
+
+        @Override
+        public boolean process(byte value) {
+            int b = value & 0xFF;
+            int code = codes[b];
```

Comment:
consider just inlining this ... `code` is only used a single time and the variable name `code` doesn't provide much additional context